src/daos_vol_dset.c

```diff
@@ -1425,7 +1425,7 @@ H5_daos_dset_open_recv_comp_cb(tse_task_t *task, void H5VL_DAOS_UNUSED *args)
             /* Open dataset */
             if (0 != (ret = daos_obj_open(
                           udata->bcast_udata.obj->item.file->coh, udata->bcast_udata.obj->oid,
-                          udata->bcast_udata.obj->item.file->flags & H5F_ACC_RDWR ? DAOS_COO_RW : DAOS_COO_RO,
+                          udata->bcast_udata.obj->item.file->flags & H5F_ACC_RDWR ? DAOS_OO_RW : DAOS_OO_RO,
```

Comment:
Does this change require a specific newer version of DAOS or have these flags been around?

Reply:
they have been around since 1.0, but i think whoever wrote this just did a typo and used the container permissions flags instead of the object specific ones. it worked before because object access was not checking object open flags, but we are implementing that now and seeing problems here.

Reply:
Ok, sounds good to me!